prow/cmd/plank/reporter.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"fmt"
+
+	prowapi "k8s.io/test-infra/prow/apis/prowjobs/v1"
+	prowjobinformer "k8s.io/test-infra/prow/client/informers/externalversions"
+	"k8s.io/test-infra/prow/config"
+	"k8s.io/test-infra/prow/config/secret"
+	"k8s.io/test-infra/prow/crier"
+	githubreporter "k8s.io/test-infra/prow/crier/reporters/github"
+	"k8s.io/test-infra/prow/flagutil"
+	"k8s.io/test-infra/prow/kube"
+)
+
+func deprecatedReporter(github flagutil.GitHubOptions, kubernetes flagutil.KubernetesOptions, dryRun bool, cfg config.Getter) (func(context.Context), error) {
+	var secretAgent secret.Agent
+	if github.TokenPath != "" {
+		if err := secretAgent.Start([]string{github.TokenPath}); err != nil {
+			return nil, fmt.Errorf("start secret agent: %w", err)
+		}
+	}
+
+	githubClient, err := github.GitHubClient(&secretAgent, dryRun)
+	if err != nil {
+		return nil, fmt.Errorf("github client: %w", err)
+	}
+
+	prowjobClientset, err := kubernetes.ProwJobClientset(cfg().ProwJobNamespace, dryRun)
+	if err != nil {
+		return nil, fmt.Errorf("prow client: %w", err)
+	}
+
+	const resync = 0
+	prowjobInformerFactory := prowjobinformer.NewSharedInformerFactoryWithOptions(prowjobClientset, resync, prowjobinformer.WithNamespace(cfg().ProwJobNamespace))
+	informer := prowjobInformerFactory.Prow().V1().ProwJobs()
+	githubReporter := githubreporter.NewReporter(githubClient, cfg, prowapi.ProwJobAgent(""))
+	controller := crier.NewController(
```

Comment:
Didn't you at some point heavily oppose linebreaking for func arguments?

Reply:
lol, not sure. Want me to switch it around?

Reply:
No, I like it this way, that's why I am surprised. The discussion we had was for the case where the function calls return is captured: https://github.com/kubernetes/test-infra/pull/15078#discussion_r344463949

Reply:
Ah, my beef there was `if err := foo()` and `; err != nil {` being on separate lines, not putting the arguments on separate lines


```golang
// fine
if bar, err := foo(); err != nil {
  return fmt.Errorf("food: %w", err)
}

// fine
something_interesting(
  with,
  lots,
  of,
  arguments)

// less excited about this
if this, that, other, err := something_interesting(
  with,
  lots,
  of,
  arguments); err != nil {
  return fmt.Errorf("less exciting: %w", err)
}
```